clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -183,4 +197,50 @@ public synchronized long lastSuccessfulUpdate() {
     public long refreshBackoff() {
         return refreshBackoffMs;
     }
+
+    /**
+     * Set state to indicate if metadata for all topics in Kafka cluster is required or not.
+     * @param needMetadaForAllTopics boolean indicating need for metadata of all topics in cluster.
+     */
+    public void needMetadataForAllTopics(boolean needMetadaForAllTopics) {
+        this.needMetadataForAllTopics = needMetadaForAllTopics;
```

Comment:
This will be called only when a pattern is subscribed. At that time yes, what you are saying makes sense. However, in subsequent metadata updates the changes in NetworkClient is still needed as metadata will have topics that matched pattern and were added in last update. Makes sense?